src/dataflow/server.rs

```diff
@@ -707,17 +693,15 @@ where
                     //     last_offset
                     // );
                     ts.push((partition_count, timestamp, offset));
-                    if last_offset == offset {
-                        // We only activate the source if the offset is the same as the last
-                        // offset as new data already triggers the source's activation
-                        let source = self
-                            .ts_source_mapping
-                            .get(&id)
-                            .expect("Id should be present");
-                        if let Some(source) = source.upgrade() {
-                            if let Some(token) = &*source {
-                                token.activate();
-                            }
+                    // We only activate the source if the offset is the same as the last
+                    // offset as new data already triggers the source's activation
+                    let source = self
+                        .ts_source_mapping
+                        .get(&id)
+                        .expect("Id should be present");
+                    if let Some(source) = source.upgrade() {
+                        if let Some(token) = &*source {
+                            token.activate();
```

Comment:
It is too late at night for me to understand why the conditional is no longer necessary. Feel free to merge if Brennan is good with it, but otherwise I will try to think harder about this when it is not so late at night.

Reply:
I was trying to be "efficient" and only triggered a source activation when fast-forward timestamp data arrived (data where last_offset == offset). In fact, there's times where I need to trigger an activation eve when last_offset!=offset. Otherwise the source will never get scheduled again (which was causing the issues we were seeing). I also had to fix this in the Kafka multi-partition case.